src/replay/exec_async.zig

```diff
@@ -11,916 +8,426 @@ const core = sig.core;
+        entries: []const Entry,
+        initial_hash: Hash,
+
+        fn run(task: *ThreadPool.Task) void {
+            const zone = tracy.Zone.init(@src(), .{ .name = "replayVerify" });
+            defer zone.deinit();
+
+            const self: *Worker = @alignCast(@fieldParentPtr("task", task));
+            defer self.future.finish();
+
+            const success = verifyPoh(
+                self.entries,
+                self.future.allocator, // cant use future's arena here as its multi-threaded.
+                self.initial_hash,
+                .{ .exit = &self.future.exit },
+            ) catch |e| switch (e) {
+                error.Exit => return,
+                error.OutOfMemory => {
+                    self.future.logger.err().log("poh verification failed with OutOfMemory");
+                    self.future.setError(error.OutOfMemory);
+                    return;
                 },
-                else => return e,
             };
-            // trySchedule will always return true, meaning the task was
-            // scheduled successfully, because the thread pool does not have a
-            // maximum number of tasks. See the `null` value passed into
-            // HomogeneousThreadPool.initBorrowed
-            assert(try self.thread_pool.trySchedule(self.allocator, .{
-                .allocator = self.allocator,
-                .logger = self.logger,
-                .committer = self.committer,
-                .svm_gateway = &self.svm_gateway,
-                .batch_index = self.batches_started,
-                .transactions = batch.transactions,
-                .results = &self.results,
-                .exit = self.exit,
-            }));
-            self.batches_started += 1;
-            tracy.plot(u32, "batches_started", @intCast(self.batches_started));
-        }
-        return null;
-    }
-};
 
-const ReplayBatchTask = struct {
-    allocator: Allocator,
-    logger: Logger,
-    svm_gateway: *SvmGateway,
-    committer: Committer,
-    batch_index: usize,
-    transactions: []const ResolvedTransaction,
-    results: *Channel(TransactionScheduler.BatchMessage),
-    exit: *Atomic(bool),
-
-    pub fn run(self: *ReplayBatchTask) !void {
-        const result = try replay.execution.replayBatch(
-            self.allocator,
-            self.svm_gateway,
-            self.committer,
-            self.transactions,
-            self.exit,
-        );
-
-        if (result == .failure) {
-            self.logger.err().logf("batch failed due to transaction error: {}", .{result.failure});
-            self.exit.store(true, .monotonic);
+            if (!success) {
+                self.future.logger.err().log("poh verification failed");
+                self.future.setError(.{ .invalid_block = .InvalidEntryHash });
+            }
         }
-        try self.results.send(.{ .batch_index = self.batch_index, .result = result });
-    }
+    };
 };
 
-test "TransactionScheduler: happy path" {
```

Comment:
Why did you remove all these tests? Can't they be adapted to work with the new approach?